### PR TITLE
python38Packages.h5py: 3.1.0 -> 3.2.1

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabaff — COMMENTED** [review]
Could you please update the license in the process? [BSD3](https://github.com/h5py/h5py/blob/master/LICENSE) and not BSD2.

**Comment by fabaff** [comment]
`hickle` want [`h5py<3.0.0`](https://github.com/telegraphic/hickle/blob/master/requirements.txt).

```bash
  ERROR: Could not find a version that satisfies the requirement h5py<3.0.0,>=2.8.0 (from hickle)
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118844` at 0f56dbe1 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.arviz</li>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.graph_nets</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4201959e478a251ac765330e443e49d4">python38Packages.hickle</a></li>
    <li><a href="https://gist.github.com/18e312a79276327931fa7aacb995ace9">python39Packages.hickle</a></li>
  </ul>
</details>
<details>
  <summary>30 packages skipped due to time constraints:</summary>
  <ul>
    <li>gpt2tc</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.caffe</li>
    <li>python38Packages.dipy</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.nilearn</li>
    <li>python38Packages.pywick</li>
    <li>python38Packages.scikit-tda</li>
    <li>python38Packages.sfepy</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>55 packages built successfully:</summary>
  <ul>
    <li>python38Packages.Keras</li>
    <li>python38Packages.annoy</li>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.bayespy</li>
    <li>python38Packages.dicom2nifti</li>
    <li>python38Packages.h5netcdf</li>
    <li>python38Packages.h5py</li>
    <li>python38Packages.h5py-mpi</li>
    <li>python38Packages.keras-applications</li>
    <li>python38Packages.meshio</li>
    <li>python38Packages.nibabel</li>
    <li>python38Packages.nipy</li>
    <li>python38Packages.nipype</li>
    <li>python38Packages.nitime</li>
    <li>python38Packages.phonopy</li>
    <li>python38Packages.pybids</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.python-openems</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.sumo</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.Keras</li>
    <li>python39Packages.annoy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.bayespy</li>
    <li>python39Packages.dicom2nifti</li>
    <li>python39Packages.h5netcdf</li>
    <li>python39Packages.h5py</li>
    <li>python39Packages.h5py-mpi</li>
    <li>python39Packages.keras-applications</li>
    <li>python39Packages.meshio</li>
    <li>python39Packages.nibabel</li>
    <li>python39Packages.nipy</li>
    <li>python39Packages.nipype</li>
    <li>python39Packages.nitime</li>
    <li>python39Packages.phonopy</li>
    <li>python39Packages.pybids</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.python-openems</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aqua</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.sumo</li>
    <li>python39Packages.worldengine</li>
    <li>visidata</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `python`.
Near pkgs/development/python-modules/h5py/default.nix:1:27:
```
  |
1 | { lib, fetchPypi, isPy27, python, buildPythonPackage, pythonOlder
  |                           ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/h5py/default.nix:2:51:
```
  |
2 | , numpy, hdf5, cython, six, pkgconfig, unittest2, fetchpatch
  |                                                   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md">duplicate-check-inputs</a>

Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- openssh-8.5p1

Near pkgs/development/python-modules/h5py/default.nix:39:3:
```
   |
39 |   checkInputs = lib.optional isPy27 unittest2 ++ [ pytestCheckHook openssh ];
   |   ^
```

Near pkgs/development/python-modules/h5py/default.nix:43:3:
```
   |
43 |   propagatedBuildInputs = [ numpy six]
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/h5py/default.nix:49:3:
```
   |
49 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by SuperSandro2000** [comment]
> `hickle` want [`h5py<3.0.0`](https://github.com/telegraphic/hickle/blob/master/requirements.txt?rgh-link-date=2021-04-08T21%3A42%3A57Z).
> 
> ```shell
>   ERROR: Could not find a version that satisfies the requirement h5py<3.0.0,>=2.8.0 (from hickle)
> ```

That was already broken before this update. Let me run nixpkgs-review and then we can decide what to do with the failures.

**Review by FRidh — APPROVED** [review]

**Comment by fabaff** [comment]
> That was already broken before this update. Let me run nixpkgs-review and then we can decide what to do with the failures.

Yes, it was. https://github.com/telegraphic/hickle/issues/143

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
11 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.arviz</li>
<li>python38Packages.clifford</li>
<li>python38Packages.dm-sonnet</li>
<li>python38Packages.graph_nets</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pymc3</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.dm-sonnet</li>
<li>python39Packages.graph_nets</li>
<li>python39Packages.optuna</li>
<li>python39Packages.rl-coach</li>
</ul>
</details>
<details>

  <summary>6 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.hickle: log was empty</li>
<li>python38Packages.nilearn: log was empty</li>
<li>python38Packages.pywick: log was empty</li>
<li>python39Packages.hickle: log was empty</li>
<li>python39Packages.nilearn: log was empty</li>
<li>python39Packages.pywick: log was empty</li>
  </ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python38Packages.qiskit: log was empty</li>
<li>python38Packages.qiskit-aqua: log was empty</li>
  </ul>
</details>
<details>
<summary>
64 packages built:
</summary>
<ul>
<li>gpt2tc</li>
<li>python38Packages.Keras</li>
<li>python38Packages.annoy</li>
<li>python38Packages.aplpy</li>
<li>python38Packages.asdf</li>
<li>python38Packages.bayespy</li>
<li>python38Packages.caffe</li>
<li>python38Packages.dicom2nifti</li>
<li>python38Packages.dipy</li>
<li>python38Packages.h5netcdf</li>
<li>python38Packages.h5py</li>
<li>python38Packages.h5py-mpi</li>
<li>python38Packages.keras-applications</li>
<li>python38Packages.meshio</li>
<li>python38Packages.nibabel</li>
<li>python38Packages.nipy</li>
<li>python38Packages.nipype</li>
<li>python38Packages.nitime</li>
<li>python38Packages.phonopy</li>
<li>python38Packages.pybids</li>
<li>python38Packages.pyregion</li>
<li>python38Packages.pytest-astropy</li>
<li>python38Packages.pytest-astropy-header</li>
<li>python38Packages.python-openems</li>
<li>python38Packages.radio_beam</li>
<li>python38Packages.reproject</li>
<li>python38Packages.sfepy</li>
<li>python38Packages.spectral-cube</li>
<li>python38Packages.sumo</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.worldengine</li>
<li>python39Packages.Keras</li>
<li>python39Packages.annoy</li>
<li>python39Packages.aplpy</li>
<li>python39Packages.asdf</li>
<li>python39Packages.bayespy</li>
<li>python39Packages.caffe</li>
<li>python39Packages.dicom2nifti</li>
<li>python39Packages.dipy</li>
<li>python39Packages.h5netcdf</li>
<li>python39Packages.h5py</li>
<li>python39Packages.h5py-mpi</li>
<li>python39Packages.keras-applications</li>
<li>python39Packages.meshio</li>
<li>python39Packages.nibabel</li>
<li>python39Packages.nipy</li>
<li>python39Packages.nipype</li>
<li>python39Packages.nitime</li>
<li>python39Packages.phonopy</li>
<li>python39Packages.pybids</li>
<li>python39Packages.pyregion</li>
<li>python39Packages.pytest-astropy</li>
<li>python39Packages.pytest-astropy-header</li>
<li>python39Packages.python-openems</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.radio_beam</li>
<li>python39Packages.reproject</li>
<li>python39Packages.sfepy</li>
<li>python39Packages.spectral-cube</li>
<li>python39Packages.sumo</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.worldengine</li>
<li>visidata</li>
</ul>
</details>
